oqsprov/oqs_kem.c

```diff
@@ -116,13 +116,41 @@ static int oqs_qs_kem_encaps_keyslot(void *vpkemctx, unsigned char *out,
         OQS_KEM_PRINTF("OQS Warning: OQS_KEM not initialized\n");
         return -1;
     }
-    *outlen = kem_ctx->length_ciphertext;
-    *secretlen = kem_ctx->length_shared_secret;
+    if (pkemctx->kem->comp_pubkey[keyslot] == NULL) {
```

Comment:
This already is a marked improvement; while we're at it, would it be prudent to also check for `pkemctx->kem->comp_pubkey != NULL`?